src/dict.h

```diff
@@ -33,8 +33,8 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#ifndef __DICT_H
-#define __DICT_H
+#ifndef DICT_H
+#define DICT_H
```

Comment:
Why rename this? It's a fairly common pattern in other code. I'm not sure if there is a strong reason to change it.

Reply:
I remember @zuiderkwast requested such similar renaming because the leading double underscore which is a reserved name by the C standard.

Reply:
I mostly meant that when we're anyway renaming things like `__REDIS_XYZ`, we can drop the leading underscores at the same time.

I don't think we want do it all over the code base. (Sorry if I said that.)

Reply:
What's the motivation to drop the leading underscores? It's just a little bit less consistent now.

Reply:
They're reserved names in C.

https://www.gnu.org/software/libc/manual/html_node/Reserved-Names.html

Reply:
I don't think it will ever be a problem, unless name macros `__ARM` or something like that.

If you prefer to keep double underscores for consistency, I can accept that.

Reply:
@lipzhu Please revert back to `__DICT_H`. If we ever want to change all these reserved names (for some kind of validator in the future) we can fix all files at the same time.

Reply:
> If you prefer to keep double underscores for consistency, I can accept that.

I had read that reserved-names doc multiple times, and never noticed `__` clause. I would prefer we keep what we have (since there isn't a conflict), but moving forward use the style that we mentioned here.

Reply:
In case you still didn't find it:

> In addition to the names documented in this manual, reserved names include all external identifiers (global functions and variables) that begin with an underscore (‘_’) and all identifiers regardless of use that begin with either two underscores or an underscore followed by a capital letter are reserved names.